src/Parser/TemplateParser.cs

```diff
@@ -138,20 +151,78 @@ private string ParseName(out int parameterIndex)
                 c = Peek();
                 if (c == '}' || c == ':' || c == ',')
                     parameterIndex = parsed;
-                _position = start;        
+                _position = start;
             }
 
             if (parameterIndex == -1)
                 _isPositional = false;
 
-            return ReadUntil(HoleDelimiters);            
+            return ReadUntil(HoleDelimiters);
         }
 
+        /// <summary>
+        /// Parse format after hole name/index
+        /// </summary>
+        /// <returns></returns>
         private string ParseFormat()
         {
             Skip(':');
-            // TODO: Escaped }} in formats?
-            return ReadUntil('}');
+            var format = ParseFormatInner();
+            return format;
+        }
+
+        /// <summary>
+        /// Parse inner of format, handle the escaped { and } in the format.
+        /// </summary>
+        /// <returns></returns>
+        private string ParseFormatInner()
+        {
+            var format = ReadUntil(TextDelimiters);
+            var c = Read();
+            if (_position == _length)
+            {
+                //end of template so done
+                return format;
+            }
+
+            switch (c)
+            {
+                case '}':
+                    {
+                        var next = Peek();
+                        if (next == '}')
+                        {
+                            //this is an escaped } and need to be added to the format.
+                            Read();
+                            format += '}' + ParseFormatInner();
```

Comment:
Maybe it's unexpected but using a string here `"}"` will give you better results